packages/utils/src/schema/sanitizeDataForNewSchema.ts

```diff
@@ -137,8 +139,14 @@ export default function sanitizeDataForNewSchema<
     });
 
     newFormData = {
-      ...data,
```

Comment:
@cwendtxealth These changes are very similar to https://github.com/rjsf-team/react-jsonschema-form/pull/3443#issuecomment-1438825962. In that PR I requested the developer to ensure that none of the other issues were regressed. Are you willing to do that legwork?

Reply:
Yeah I should be able to validate those issues if they have reproducible steps

Reply:
Just from some initial testing it does look like removing some undefined fields causes some issues with how `getDefaultFormState` fills out `formData`. Might need to look at a different place to remove the undefined fields.

Reply:
@cwendtxealth I'm on vacation for a week starting today. My thoughts are that, perhaps we add another flag to the `experimental_defaultFormStateBehavior` that deals with this behavior somehow.

Reply:
Maybe we talk about this at one of the Friday meetings?

Reply:
That sounds good!